api/axis.py

```diff
@@ -211,10 +211,10 @@ def update_licensepools_for_identifiers(self, identifiers):
         # k books are the identifiers in `remainder`. These books have
         # been removed from the collection without us being notified.
         for removed_identifier in remainder:
-            pool = removed_identifier.licensed_through
+            pool = identifier.licensed_through_collection(self.collection)
```

Comment:
removed_identifier.licensed_through now returns a list. We only want the license for this identifier through this collection.